frappe/database/db_manager.py

```diff
@@ -70,16 +70,16 @@ def restore_database(target, source, user, password):
 
 		command = (
 			"{pipe} mysql -u {user} -p{password} -h{host} "
-			+ ("-P{port}" if frappe.db.port else "")
+			+ ("-P{port}" if frappe.conf.db_port else "")
 			+ " {target} {source}"
 		)
 		command = command.format(
 			pipe=pipe,
 			user=esc(user),
 			password=esc(password),
-			host=esc(frappe.db.host),
+			host=esc(frappe.conf.db_host or "127.0.0.1"),
```

Comment:
_Leaf Defaulting_ can be removed after we can enjoy the guarantees of  https://github.com/frappe/frappe/pull/21577